run.py

```diff
@@ -33,17 +37,80 @@ def _get_opt_parser(self):
         parser.add_option('-p', '--pass', dest='password', default=None)
         return parser
 
-    def _get_connection(self):
+    def _get_connection(self, hostname, username, password):
         client = paramiko.SSHClient()
         client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
-        client.connect(hostname=self.options.host, username=self.options.user,
-                       password=self.options.password)
+        client.connect(hostname=hostname, username=username,
+                       password=password)
         return client
 
-    def _execute_command(self, command):
-        (stdin, stdout, stderr) = self.connection.exec_command(command)
+    def _execute_command(self, connection, command):
+        (stdin, stdout, stderr) = connection.exec_command(command)
         return (stdout, stderr)
 
+    def troubleshoot_mon(self):
+        command = 'sudo ceph health'
```

Comment:
We just need to key off the ceph version that is running.  If it's
infernalis or newer than we use the ceph users.  Older than infernalis and
it requires root.  See here:
https://github.com/openstack/charm-ceph-mon/blob/89ac19a2571da94f37a738335a6fb6f43a744255/hooks/charmhelpers/contrib/storage/linux/ceph.py#L925

On Thu, May 19, 2016 at 9:21 AM, Kumar Rishabh <notifications@github.com>
wrote:

> In run.py
> <https://github.com/CanonicalLtd/DiagCephCluster/pull/1#discussion_r63909734>
> :
>
> >          return (stdout, stderr)
> >
> > +    def troubleshoot_mon(self):
> > +        command = 'sudo ceph health'
>
> So to support different versions shouldn't I run ceph commands both
> without sudo and with sudo? Or just support the one which doesn't require
> root?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/CanonicalLtd/DiagCephCluster/pull/1/files/59a5b986b3412dbddf69c81ff5df6ff2c8b9e297#r63909734>
>